src/Microsoft.OData.Core/ODataCollectionWriterCore.cs

```diff
@@ -860,6 +859,26 @@ await this.InterceptExceptionAsync((thisParam) => thisParam.EndPayloadAsync())
             }
         }
 
+        /// <summary>
+        /// Flush asynchronously - actual implementation
+        /// </summary>
+        /// <returns>A task instance that represents the asynchronous operation.</returns>
+        private async Task FlushImplementationAsync()
+        {
+            // Make sure we switch to writer state Error if an exception is thrown during flushing.
+            try
+            {
+                await this.FlushAsynchronously()
+                    .ConfigureAwait(false);
+            }
+            catch
+            {
+                this.ReplaceScope(CollectionWriterState.Error, null);
+

```

Comment:
I think we need to be consistent on if we need this line space before `throw;`

Reply:
@KenitoInc What would you prefer?

Reply:
Having a linespace before `throw;`

Reply:
Me too. It's why I added one here